src/psyclone/tests/psyir/symbols/symboltable_test.py

```diff
@@ -1643,3 +1646,232 @@ def test_rename_symbol():
+    b_2 = subroutine.symbol_table.lookup('b_2')
+    not_used1 = subroutine.symbol_table.lookup('not_used1')
+    assert "not_used2" not in subroutine.symbol_table
+    assert not isinstance(not_used1, DataSymbol)
+    assert isinstance(a_1.interface, ImportInterface)
+    assert not isinstance(a_1, DataSymbol)
+    assert isinstance(a_2.interface, ImportInterface)
+    assert not isinstance(a_2, DataSymbol)
+    assert a_2.visibility == Symbol.Visibility.PRIVATE
+    assert isinstance(b_1.interface, UnresolvedInterface)
+    assert not isinstance(b_1, DataSymbol)
+    assert isinstance(b_2.interface, UnresolvedInterface)
+    assert not isinstance(b_2, DataSymbol)
+
+    # Set up include_path to import the proper modules
+    Config.get()._include_paths = [str(tmpdir)]
+
+    # Resolve symbols that are in a_mod inside the subroutine
+    subroutine.symbol_table.resolve_imports([
+            subroutine.symbol_table.lookup('a_mod')])
+
+    # This will resolve a_1 information
+    assert isinstance(a_1, DataSymbol)
+    assert a_1.datatype.intrinsic.name == 'INTEGER'
+    # but will keep the ImportInterface
+    assert isinstance(a_1.interface, ImportInterface)
+
+    # The other symbols (including a_2) are unchanged
+    assert "not_used2" not in subroutine.symbol_table
+    assert isinstance(a_2.interface, ImportInterface)
+    assert not isinstance(a_2, DataSymbol)
+    assert isinstance(b_1.interface, UnresolvedInterface)
+    assert not isinstance(b_1, DataSymbol)
+    assert isinstance(b_2.interface, UnresolvedInterface)
+    assert not isinstance(b_2, DataSymbol)
+
+    # Now resolve all found containers (this will not fail for the
+    # unavailable c_mod)
+    subroutine.symbol_table.resolve_imports()
+
+    # b_1 and b_2 should have relevant info now
+    assert isinstance(b_1, DataSymbol)
+    assert b_1.datatype.intrinsic.name == 'INTEGER'
+    assert b_1.constant_value.value == "10"
+    assert isinstance(b_2, DataSymbol)
+    assert isinstance(b_2.datatype, UnknownFortranType)
+    # Their interface is also updated
+    assert isinstance(b_1.interface, ImportInterface)
+    assert b_1.interface.container_symbol == \
+           subroutine.symbol_table.lookup('b_mod')
+    assert isinstance(b_2.interface, ImportInterface)
+    assert b_2.interface.container_symbol == \
+           subroutine.symbol_table.lookup('b_mod')
+    # not_used1 and not_used2 exist and are DataSymbols now
+    assert isinstance(not_used1, DataSymbol)
+    assert isinstance(subroutine.symbol_table.lookup('not_used2'), DataSymbol)
+
+    # a_2 is not yet resolved because if is from another symbol table,
+    # resolve that symbol table too
+    assert not isinstance(a_2, DataSymbol)
+    subroutine.parent.symbol_table.resolve_imports()
+    # In this case check that the visibility stays PRIVATE
+    assert isinstance(a_2, DataSymbol)
+    assert a_2.visibility == Symbol.Visibility.PRIVATE
+
+    # Clean up the config instance
```

Comment:
monkeypatch will remove the need for this (and may also be better if the test should fail).

Reply:
Done